app/Http/Middleware/RequireTwoFactorAuthentication.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Pterodactyl\Http\Middleware;
+
+use Closure;
+use Krucas\Settings\Settings;
+use Prologue\Alerts\AlertsMessageBag;
+
+class RequireTwoFactorAuthentication
+{
+    const NOBODY = 0;
+    const ADMINISTRATORS = 1;
+    const USERS = 2;
+    const EVERYBODY = 3;
+
+    /**
+     * @var \Prologue\Alerts\AlertsMessageBag
+     */
+    protected $alert;
+
+    /**
+     * @var \Krucas\Settings\Settings
+     */
+    protected $settings;
+
+    /**
+     * RequireTwoFactorAuthentication constructor.
+     *
+     * @param \Prologue\Alerts\AlertsMessageBag $alert
+     * @param \Krucas\Settings\Settings         $settings
+     */
+    public function __construct(Settings $settings, AlertsMessageBag $alert)
+    {
+        $this->settings = $settings;
+        $this->alert = $alert;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        // Ignore non-users
+        if (! auth()->check()) {
+            return $next($request);
+        }
+
+        // All TOTP related routes
+        $validRoutes = [
```

Comment:
This variable name is a bit confusing, maybe `$ignoreRoutes` or do what Laravel often does for things and have `protected $ignore = [...routes];`.

Reply:
`protected $ignore = []` sounds good to me